git-push-each

```diff
@@ -1,9 +1,17 @@
 #!/bin/sh
 
 usage="USAGE
-    $0 [--clean] <remote> <branch-prefix>
+    $0 [--clean] [<remote> [<branch-prefix>]]
     push each commit from stdin to its own branch.
 
+DESCRIPTION
+    Read commit <sha1> hashes from stdin, one per line, and
+    for each one, push a branch to <remote>.
+    The branch is named following the <branch-prefix>-<sha1>
+    pattern and points to <sha1>.
+    By default <branch-prefix> is 'gpe' and <remote> is
```

Comment:
i had the prefix as `gpe-` so that people could change it to `whatever/` if they prefer some other delimiter. (some tools are dumb and can get confused if branch names have `/`, hence i defaulted to `-`, though `/` is nicer)

this probably isn't a big deal.